### PR TITLE
Fixes #59: incorrect processing of dirs in paths

### LINKED ISSUE #59: directories specified in the array "paths" in yml configuration file are not processed correctly
part of my yml configuration file:

``` yml
hearsay_require_js:
  initialize_template: HearsayRequireJSBundle::initialize.html.twig
  base_dir: "%kernel.root_dir%/Resources/public/js"
  base_url: "compiled/js/require"
  paths:
    global: "%kernel.root_dir%/Resources/global/js"
```

Actual result in generated config:

``` javascript
var require={"baseUrl":"/compiled/js/require","locale":"en","paths":{"global":"/compiled/js/require/global/js"
```

Expected result in generated config:

``` javascript
var require={"baseUrl":"/compiled/js/require","locale":"en","paths":{"global":"/compiled/js/require/global"
```

As you can see, js subdirectory is redundant here: 
/compiled/js/require/global__/js__

### PR DESCRIPTION
**Pull request by apsavin**

### CODE REVIEW

File: Configuration/NamespaceMapping.php [hunk 1]
```diff
@@ -66,8 +66,12 @@ public function getModulePath($filename)
 
         foreach ($this->namespaces as $namespace => $realPath) {
             if (strpos($filePath, $realPath) === 0) {
-                $basename   = $this->getBaseName($filePath, $realPath);
-                $modulePath = $this->basePath . '/' . $namespace . '/' . $basename;
+                $modulePath = $this->basePath . '/' . $namespace;
+
+                if (!is_dir($filePath)) {
```
Comment (ihortymoshenko) [inline Configuration/NamespaceMapping.php]: `!is_dir($filePath)` -> `is_file($filePath)` - it will be easier to understand.

**Comment by ihortymoshenko** [comment]
@apsavin, looks like I made ​​the mistake. Thanks!

File: Configuration/NamespaceMapping.php (hunk 1 above)
Comment (apsavin) [inline Configuration/NamespaceMapping.php]: yeah.

Comment (apsavin) [inline Configuration/NamespaceMapping.php]: yeah.

File: Configuration/NamespaceMapping.php [hunk 2]
```diff
@@ -66,8 +66,12 @@ public function getModulePath($filename)
 
         foreach ($this->namespaces as $namespace => $realPath) {
             if (strpos($filePath, $realPath) === 0) {
-                $basename   = $this->getBaseName($filePath, $realPath);
-                $modulePath = $this->basePath . '/' . $namespace . '/' . $basename;
+                $modulePath = $this->basePath . '/' . $namespace;
+
+                if (is_file($filePath)) {
+                    $basename   = $this->getBaseName($filePath, $realPath);
```
Comment (ihortymoshenko) [inline Configuration/NamespaceMapping.php]: Please remove the temporary variable.

**Comment by apsavin** [comment]
It's really important for me to have "paths" option working . What can I do to speed up fixing of this issue?

**Comment by ihortymoshenko** [comment]
@apsavin, done. Thanks for the contribution!

**Pull request merged**